### PR TITLE
Add span_extratcor for context aware environments

### PR DESCRIPTION
**Pull request by mohabusama**
Add span_extratcor for context aware environments

+ Add more tests - 100%
+ Add docstrings

### CODE REVIEW

File: opentracing_utils/decorators.py [hunk 1]
```diff
@@ -4,14 +4,40 @@
 
 
 def trace(component=None, operation_name=None, tags=None, pass_span=False, inspect_stack=True, ignore_parent_span=False,
-          **kwargs):
-    """Opentracing docrator"""
+          span_extractor=None, **kwargs):
+    """
+    Opentracing tracer docrator. Attempts to extract parent span and create a new span for the decorated function.
```
Comment (elgris) [inline opentracing_utils/decorators.py]: Typo. `docrator` --> `decorator`

File: opentracing_utils/span.py [hunk 2]
```diff
@@ -5,12 +5,21 @@
...
+
+
+def get_new_span(f, operation_name=None, inpsect_stack=True, ignore_parent_span=False, span_extractor=None, **kwargs):
     parent_span = None
-    span_arg_name = None
+    span_arg_name = DEFAULT_SPAN_ARG_NAME
 
     if not ignore_parent_span:
-        span_arg_name, parent_span = get_parent_span(inpsect_stack=inpsect_stack, **kwargs)
+        span_arg_name = DEFAULT_SPAN_ARG_NAME
```
Comment (elgris) [inline opentracing_utils/span.py]: `span_arg_name` is already `DEFAULT_SPAN_ARG_NAME ` here

Comment (mohabusama) [reply opentracing_utils/span.py]: Done.

File: opentracing_utils/decorators.py (hunk 1 above)
Comment (mohabusama) [reply opentracing_utils/decorators.py]: Done

File: tests/test_trace.py [hunk 3]
```diff
@@ -210,3 +213,71 @@ def f2():
...
+    @trace(span_extractor=extractor)
+    def f1():
+        pass
+
+    with other_span:
+        # other_span could be ignored if extractor returned a span!
+        f1()
+
+    if return_span:
+        recorder.spans[0].context.trace_id == test_span.context.trace_id
```
Comment (elgris) [inline tests/test_trace.py]: do you need `assert` for these statements?

Comment (mohabusama) [reply tests/test_trace.py]: yes. nice catch *facepalm*

**Comment by mohabusama** [comment]
👍

**Comment by elgris** [comment]
👍

**Comment by mohabusama** [comment]
👍

**Comment by mohabusama** [comment]
👍

**Comment by lmineiro** [comment]
👍

**Pull request merged**